### PR TITLE
Remove vowels from rand.String, to avoid 'bad words'

### PR DESCRIPTION
**Pull request by thockin**
As reported by users via email.

@aronchick

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37225)

### CODE REVIEW

**Comment by saad-ali** [comment]
Bug fixes ok for post-code-freeze merge.

**Comment by Philmod** [comment]
Nice!

**Comment by dims** [comment]
Wanna re-add your lgtm @saad-ali ?

**Comment by saad-ali** [comment]
@k8s-bot unit test this

**Comment by saad-ali** [comment]
@thockin `Jenkins unit/integration` might be a real failure.

**Comment by thockin** [comment]
I can't see how - I threw some additional debugging in that test and I can't see why it is failing, much less due to this PR..will keep digging

**Comment by thockin** [comment]
I have repushed with more debug - I can't see why reflect.DeepEqual is failing...

**Comment by dims** [comment]
@thockin - Here's what you need :) the test uses rand.Seed(1) and the values generated are hard coded in the test.
http://paste.openstack.org/show/590068/

**Comment by thockin** [comment]
That's just awful... But thanks!

I'm still very confused - the failing test line is a reflect.DeepEqual, but
when I dump the objects being compared they appear identical.

**Comment by dims** [comment]
@thockin right, i noticed that onion but did not peel it :)

**Comment by thockin** [comment]
@mikedanese bazel is complaining about not finding spew, but it is present in vendor/  - PTAL?

**Comment by thockin** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by mikedanese** [comment]
looking...

**Comment by mikedanese** [comment]
I think you needed to just rerun update-bazel.sh to fix the BUILD file. Feel free to squash the commit I appended.

**Comment by dims** [comment]
@k8s-bot test this

**Comment by dims** [comment]
@bprashanth @saad-ali one of you please lgtm this?

File: pkg/util/rand/rand.go [hunk 1]
```diff
@@ -72,12 +70,16 @@ func Perm(n int) []int {
 	return rng.rand.Perm(n)
 }
 
-// String generates a random alphanumeric string n characters long.  This will
-// panic if n is less than zero.
+// We omit vowels from the set of available characters to reduce the chances
+// of "bad words" being formed.
+var alphanums = []rune("bcdfghjklmnpqrstvwxz0123456789")
```
Comment (bprashanth) [inline pkg/util/rand/rand.go]: well we still have 3,1 and 0

**Comment by bprashanth** [comment]
LGTM

**Comment by thockin** [comment]
I don't understand - this is milestone 1.5, and the queue seems empty.  Why is the queue saying "Milestone is for a future release and cannot be merged"

@saad-ali

**Comment by dims** [comment]
@thockin total freeze till monday morning - https://groups.google.com/d/msg/kubernetes-dev/n-vqlX-HHSM/_OflBuu_CAAJ

**Comment by thockin** [comment]
Roger!  Missed that.  Thanks.

On Sat, Nov 26, 2016 at 11:20 AM, Davanum Srinivas <notifications@github.com

**Comment by dims** [comment]
@k8s-bot test this

**Comment by dims** [comment]
@k8s-bot test this issue: #IGNORE

**Comment by dims** [comment]
@k8s-bot kubemark e2e test this

**Comment by dims** [comment]
@k8s-bot cvm gke e2e test this

**Comment by dims** [comment]
@k8s-bot node e2e test this

**Comment by dims** [comment]
@k8s-bot cri node e2e test this

**Comment by krzyzacy** [comment]
assume https://github.com/kubernetes/test-infra/pull/1215 would fix this

File: pkg/util/rand/rand.go (hunk 1 above)
Comment (bryan-croft) [reply pkg/util/rand/rand.go:75]: I guess we need a character set or function that properly considers "leet speak" encoded "bad words". $DEITY forbid that someone be confronted with a 'n0sh1t' Pod or something.

**Pull request merged**